src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,133 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int STORAGE_DEFAULT_CAPACITY = 10;
+    private static final Object[] DEFAULTCAPACITY_EMPTY_ELEMENTSDATA = {};
+    private Object[] elementsData;
+    private int size;
+
+    public ArrayList() {
+        this.elementsData = new Object[STORAGE_DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == elementsData.length) {
+            elementsData = grow();
```

Comment:
```suggestion
            grow();
```
this method shouldn't return anything